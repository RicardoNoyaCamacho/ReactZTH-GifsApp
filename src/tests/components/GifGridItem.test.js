import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Prueba en <GifGridItem/>", () => {
  const title = "Un titulo";
  const url = "http://www.url.com";

  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("de de mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("Debe de de tener la imagen igual al url y alt de props", () => {
    const img = wrapper.find("img");
    // console.log(img.props());
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("debe de tener animate__fadeIn ", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");

    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
