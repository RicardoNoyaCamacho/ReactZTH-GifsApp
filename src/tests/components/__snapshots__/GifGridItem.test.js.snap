// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGridItem/> de de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="http://www.url.com"
  />
  <p>
    Un titulo
  </p>
</div>
`;
